src/DonorProfiles/resources/js/app/components/donation-table/index.js

```diff
@@ -0,0 +1,78 @@
+import { Fragment, useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import Table from '../table';
+import DonationRow from './donation-row';
+
+import './style.scss';
+
+const DonationTable = ( { donations, perPage } ) => {
+	const [ page, setPage ] = useState( 1 );
+
+	let donationRows = [];
+	let start = 0;
+	let end = perPage;
+	let lastPage = 1;
+
+	if ( donations ) {
+		start = ( page - 1 ) * perPage;
+		end = start + perPage <= Object.entries( donations ).length ? start + perPage : Object.entries( donations ).length;
+		lastPage = Math.ceil( Object.entries( donations ).length / perPage );
```

Comment:
`Object.entries( donations ).length` is referenced multiple places and should probably be extracted to a variable.

Reply:
Done, ref: https://github.com/impress-org/givewp/pull/5455/commits/bba93c575b2cc638f1eca770feabb8a39666db0c